clientconn_test.go

```diff
@@ -181,6 +180,48 @@ func TestDialWaitsForServerSettings(t *testing.T) {
 
 }
 
+func TestDialWaitsForServerSettingsAndFails(t *testing.T) {
+	defer leakcheck.Check(t)
+	server, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		t.Fatalf("Error while listening. Err: %v", err)
+	}
+	done := make(chan struct{})
+	dialDone := make(chan struct{})
+	numConns := 0
+	go func() { // Launch the server.
+		defer func() {
+			close(done)
+		}()
+		for {
+			conn, err := server.Accept()
+			if err != nil {
+				break
+			}
+			numConns++
+			defer conn.Close()
+		}
+		<-dialDone // Close conn only after dial returns.
```

Comment:
This doesn't seem to do anything AFAICT? I think you can entirely remove the variable.

Reply:
Done